tests/ga/test_update.py

```diff
@@ -2073,6 +2083,166 @@ def reload_conf(url, mock_wire_data):
                                     'Discovered new {0} upgrade WALinuxAgent-99999.0.0.0; Will upgrade on or after'.format(
                                         AgentUpgradeType.Normal) in msg]), 0, "Error message not propagated properly")
 
+    def test_it_should_download_only_requested_version_if_available(self):
+        data_file = mockwiredata.DATA_FILE.copy()
+        data_file["ext_conf"] = "wire/ext_conf_requested_version.xml"
+        with self.__get_update_handler(test_data=data_file) as (update_handler, mock_telemetry):
+            with patch.object(conf, "get_enable_ga_versioning", return_value=True):
+                update_handler.run(debug=True)
+
+            self.__assert_exit_code_successful(update_handler.exit_mock)
+            upgrade_event_msgs = [kwarg['message'] for _, kwarg in mock_telemetry.call_args_list if
+                                  'Agent upgrade discovered, updating to WALinuxAgent-9.9.9.10 -- exiting' in kwarg[
+                                      'message'] and kwarg['op'] == WALAEventOperation.AgentUpgrade]
+            self.assertEqual(1, len(upgrade_event_msgs), "Agent not upgraded properly")
```

Comment:
if this fails, we'll get something like '1 != 0 Agent not upgraded properly" 

this should be improved.

maybe 

assertTrue( 1 == len(...), "Did not find the event indicating that the agent was upgraded. Got: {0}", etc)

Reply:
good point, I actually faced this problem when authoring these tests lol.
Let me update

Reply:
Done